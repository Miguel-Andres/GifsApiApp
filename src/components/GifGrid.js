// import React, { useState, useEffect } from "react";
// import { getGifs } from "../helpers/getGifs";
  import GifGridItem from "./GifGridItem";

import { useFetchGifs } from "../hooks/useFetchGifs";

export default function GifGrid({ category }) {


   const {loading ,data }= useFetchGifs(category)
  console.log(data)

  

  return (
    <>
      <h2 className=" animate__animated animate__fadeIn">{category}</h2>

      {loading && <p className="animate__animated animate__flash">Cargando</p> }

       <div className="card-grid">
        {data.map((item) => (
          <GifGridItem key={item.id} {...item} />
        ))}
      </div> 
    </>
  );
}
