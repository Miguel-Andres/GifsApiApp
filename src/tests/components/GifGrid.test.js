import { shallow } from "enzyme"
import GifGrid from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs"
jest.mock("../../hooks/useFetchGifs")


describe('test a componente <GifGrdi/>', () => { 

    
    
    test("debe mostrarse correctamente el componente",()=>{
        
        useFetchGifs.mockReturnValue({
            data :[],
            loading:true
        })

        const category = "jordan"
        const wrapper = shallow(<GifGrid category={category}/>)

        expect(wrapper).toMatchSnapshot()
    })

    test('debe de mostrar item cuando se llama a useFetchGifs', () => { 

        const gifs = [
            {id:"abc",url:"http://Api.com",title:"anything"},
            {id:"abc",url:"http://Api.com",title:"anything"}

        ]
        useFetchGifs.mockReturnValue({
            data :gifs,
            loading:false 
        })

        const category = "jordan"
        const wrapper = shallow(<GifGrid category={category}/>)

        // expect(wrapper).toMatchSnapshot()
        expect(wrapper.find("p").exists()).toBe(false)
        expect(wrapper.find("GifGridItem").length).toBe(gifs.length)
     
     })
 })