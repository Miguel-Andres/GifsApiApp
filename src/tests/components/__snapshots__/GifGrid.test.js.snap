// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test a componente <GifGrdi/> debe mostrarse correctamente el componente 1`] = `
<Fragment>
  <h2
    className=" animate__animated animate__fadeIn"
  >
    jordan
  </h2>
  <p
    className="animate__animated animate__flash"
  >
    Cargando
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
